python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -598,88 +598,256 @@ def __setattr__(self, name, value):
         return _FastSlowAttribute("__setattr__").__get__(self)(name, value)
 
     def __add__(self, other):
```

Comment:
Between this PR and #15695, we should proxy all the special methods in the Python object model -- and I think we should be able to add these methods dynamically rather than spell out every single one. Something like this (untested):

```python
for special_method in ["__add__", "__radd__", ...]:
    def special_method_impl(self, other):
        return _fast_slow_function_call(
            lambda x, y: getattr(x, special_method, lambda: NotImplemented)(y),
            self,
            other,
        )[0]
    setattr(_FastSlowProxy, special_method, special_method_impl)
```

Reply:
Same idea for unary ops like `__neg__`, `__abs__`, `__int__`, `__float__`.

Reply:
Moved all to `SPECIAL_METHODS` that will accomplish the same.

Reply:
Done, looks like unary ops are handled in SPECIAL_METHODS but those seem to be skewed towards CPU only ops, I'll open a separate issue to tackle those.